opengrok-indexer/src/main/java/org/opengrok/indexer/util/LineBreaker.java

```diff
@@ -65,65 +63,52 @@ public void reset(StreamSource src, ReaderWrapper wrapper)
             throw new IllegalArgumentException("`src' is null");
         }
 
-        length = 0;
-        lineOffsets = null;
-
-        try (InputStream in = src.getStream();
-            Reader rdr = IOUtils.createBOMStrippedReader(in,
-                StandardCharsets.UTF_8.name())) {
-            Reader intermediate = null;
-            if (wrapper != null) {
-                intermediate = wrapper.get(rdr);
-            }
-
-            try (BufferedReader brdr = new BufferedReader(
-                    intermediate != null ? intermediate : rdr)) {
-                reset(brdr);
-            } finally {
-                if (intermediate != null) {
-                    intermediate.close();
-                }
-            }
-        }
+        SplitterUtil.reset(this::reset, src, wrapper);
     }
 
     private void reset(Reader reader) throws IOException {
+        length = 0;
+        lineOffsets = null;
+
         List<Integer> newOffsets = new ArrayList<>();
         newOffsets.add(0);
 
         int c;
         while ((c = reader.read()) != -1) {
             ++length;
-            switch (c) {
-                case '\r':
-                    c = reader.read();
-                    if (c == -1) {
-                        newOffsets.add(length);
-                        break;
-                    } else {
+
+            redo_c:
+            while (true) {
```

Comment:
seriously?
do we want to go for endless cycles with goto?
back to Basic? :-D

(yes, it's easy for me to complain, but this raises eyebrows, so there has to be a better way)
(the first thing that came to my mind was a jflex lexer, but not sure if we don't have the same problem with line # there too)

Reply:
couldn't resist to ping @kahatlen here as the EOL expert ;-)
(hey Knuth, how are you these days?)

Reply:
> seriously?
> do we want to go for endless cycles with goto?
> back to Basic? :-D
> 
> (yes, it's easy for me to complain, but this raises eyebrows, so there has to be a better way)

You've never hand-written a scanner? 😄 But you're right it's generally not a good idea; imagine code reviewing a hand-written `PerlXref`. But this is a teeny one, and look at that clear `break` at the end of the inner.... (Anyway search for `while (true)` in opengrok....)

> (the first thing that came to my mind was a jflex lexer, but not sure if we don't have the same problem with line # there too)

I wrote these classes originally when it was a big pain to add additional .lex files, especially in packages where they weren't already used. But now we can add them anywhere easily (e.g. `XrefSourceTransformer`). Also I confess if I had used JFlex originally I wouldn't have had the deficient handling that I'm updating here....

How about if we decide at some point to update OpenGrok `STANDARD_EOL` as we probably should do, then I rewrite these to use JFlex? But for now I do think I'm updating and testing sufficiently in this patch.

Reply:
OK I spent some time and replaced with a couple of new JFlex scanners